esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -270,19 +270,41 @@ void WsWuInfo::readWorkunitComponentLogs(const char* outFile, unsigned maxLogRec
 
     while (logreader->readLogEntries(logcontent.clear(), recsRead))
     {
-        if (logFormat == LOGACCESS_LOGFORMAT_json && totalRecsRead > 0 && recsRead > 0 )
+        if (logFormat == LOGACCESS_LOGFORMAT_json && totalRecsRead > 0 && recsRead > 0)
         {
-            //logcontent.insert(0,",");
-            outIOS->write(1, ",");
+            writeCharToStream(*outIOS, ',');
         }
         totalRecsRead += recsRead;
         outIOS->write(logcontent.length(), logcontent.str());
     }
 
-    if (logFormat == LOGACCESS_LOGFORMAT_json)
-        outIOS->write(2, "]}");
-    else if (logFormat == LOGACCESS_LOGFORMAT_xml)
-        outIOS->write(8, "</lines>");
+    if (totalRecsRead == maxLogRecords) //Warn of possible truncation
```

Comment:
question: could request maxLogRecords+1, but truncate to maxLogRecords. That way we (and the consumer) would know for sure that it exceeded the limit / that there are more logs (even if it were 1)., but it would be unambiguous that over maxLogRecords at that point.

Reply:
Yeah interesting proposal, my concern is if the plus one forces an extra fetch (preventable, but then we still wouldn't be able to determine truncation). 
The trade off is one more potential fetch if the maxLogRecords % pagesize == 0 AND we happen to encounter exactly maxlogrecords. Tradeoff is we warn of possible truncation.
It seems to me non-deterministic warning is better.